File: NC/src/CentralDifference.cpp
#include "../include/CentralDifference.hpp"

CentralDifference::CentralDifference(function<double(double)> f, double h)
    : Differentiation(f, h) {}

double CentralDifference::derivative(double x) const {
    return (func(x + h) - func(x - h)) / (2.0 * h);
}

File: Numerical_Computing_2/src/differentiation/central_difference.py
from .numerical_differentiation import NumericalDifferentiation


class CentralDifference(NumericalDifferentiation):
    def derivative(self, x: float) -> float:
        return (self.func(x + self.h) - self.func(x - self.h)) / (2.0 * self.h)

    @property
    def method_name(self) -> str:
        return "Central Difference"
